From 715324bb99d0cd962b8d7bc7e78ae29e8b0f11ba Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 05 Sep 2021 20:27:51 +0300
Subject: [PATCH] Revert "[SW-47873] comms: fw cfg skip"

This commit is wrong and should have not been merged.

This reverts commit ed9c76b70fd8d6eb01f04e92607b593dfb3d3c01.

Change-Id: I5892438d197aa7963d321722ae742fa3f8f8a503
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index dc6d97d..5ee5748 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -52,7 +52,6 @@
 
 static struct hl_rsp_err pending_error;
 static u64 preboot_en_caps;
-static u8 fw_cfg_skip;
 
 static struct cmd_module_ops cmd_ops[] = {
 	{	/* PCI module */
@@ -484,16 +483,3 @@
 	return 1;
 }
 
-void preboot_set_fw_cfg_skip(u8 skip)
-{
-	fw_cfg_skip = skip;
-}
-
-void preboot_finalize_uboot_override(void)
-{
-	if (fw_cfg_skip)
-		boot_info_set(HL_BOOT_INFO_FW_CFG_SKIP,
-			      &fw_cfg_skip,
-			      sizeof(fw_cfg_skip));
-
-}
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 18ee402..886d5e0 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -111,11 +111,6 @@
 		.use_calib = false, .calib_offset = 8, .hl_tmp_data = { 0 }}
 };
 
-bool hl_fw_cfg_skip_support(void)
-{
-	return true;
-}
-
 void hl_ts_data_init(void)
 {
 	u32 pll_freq;
diff --git a/include/preboot.h b/include/preboot.h
index c5a50d8..08ca78a 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -212,8 +212,6 @@
 void preboot_init_pcie_phy(void);
 void preboot_engines_init(void);
 void preboot_clear_wd_gpio(void);
-void preboot_set_fw_cfg_skip(u8 skip);
-void preboot_finalize_uboot_override(void);
 
 void ppboot_init_stack_mon_dev_specific(void);
 void ppboot_check_stack_mon_dev_specific(void);
diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index 0fcaae0..e60bd1d 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -1010,7 +1010,6 @@
 	preboot_finalize_rma_tables();
 	preboot_finalize_fw_upd_closure();
 	preboot_finalize_enabled_caps();
-	preboot_finalize_uboot_override();
 	boot_info_finalize();
 	preboot_start_image(startaddr);
 }
diff --git a/lib/preboot/module_lkd.c b/lib/preboot/module_lkd.c
index 37804cc..7946254 100644
--- a/lib/preboot/module_lkd.c
+++ b/lib/preboot/module_lkd.c
@@ -95,20 +95,6 @@
 	return err;
 }
 
-__weak bool hl_fw_cfg_skip_support(void) { return false; }
-
-static int module_lkd_msg_fw_cfg_skip(struct lkd_fw_comms_msg *mcomms)
-{
-	if (!hl_fw_cfg_skip_support() && preboot_security_is_enabled()) {
-		preboot_puts("fw_cfg_skip is not supported/enabled\n");
-		return -1;
-	}
-
-	preboot_set_fw_cfg_skip(mcomms->fw_cfg_skip);
-
-	return 0;
-}
-
 static int module_lkd_msg_process(struct lkd_fw_comms_msg *mcomms)
 {
 	int err = 1;
@@ -116,10 +102,7 @@
 	switch (mcomms->header.type) {
 	case HL_COMMS_RESET_CAUSE_TYPE:
 		err = module_lkd_msg_reset_cause(mcomms);
-		break;
-	case HL_COMMS_FW_CFG_SKIP_TYPE:
-		err = module_lkd_msg_fw_cfg_skip(mcomms);
-		break;
+		/* no break */
 	default:
 		break;
 	}
